dicthash/dicthash.py

```diff
@@ -122,11 +122,14 @@ def _generate_string_from_dict(d, blacklist, whitelist, prefix=''):
     if blacklist is not None:
         whitelist = set(whitelist).difference(blacklist)
     # Sort whitelist according to the keys converted to str
-    return ''.join(_unpack_value(d[key],
-                                 whitelist=filter_blackwhitelist(whitelist, key),
-                                 blacklist=filter_blackwhitelist(blacklist, key),
-                                 prefix=prefix + str(key)) for
-                   key in sorted(filter_blackwhitelist(whitelist, None), key=str))
+    if len(whitelist) > 0:
+        return ''.join(_unpack_value(d[key],
+                                     whitelist=filter_blackwhitelist(whitelist, key),
+                                     blacklist=filter_blackwhitelist(blacklist, key),
+                                     prefix=prefix + str(key)) for
+                       key in sorted(filter_blackwhitelist(whitelist, None), key=str))
+    else:
+        return ''
```

Comment:
shouldn't this case at least give a warning, since most likely something went wrong?

Reply:
I wouldn't say that something went wrong. This `else`-Branch covers the case that you excluded all keys from your dictionary or a sub-dictionary of it.
This might be a sensible use-case.